mmengine/optim/optimizer/builder.py

```diff
@@ -146,7 +146,7 @@ def register_bitsandbytes_optimizers() -> List[str]:
                 'PagedAdamW8bit', 'LAMB8bit', 'LARS8bit', 'RMSprop8bit',
                 'Lion8bit', 'PagedLion8bit', 'SGD8bit'
         ]:
-            _optim = getattr(bnb.optim, module_name)
```

Comment:
Register a None optimizer seems weird, maybe we should skip registering the optimizer if it does not exist

Reply:
If `_optim` is `None`, the result of `inspect.isclass(_optim)` is `False` and the registering will be skipped
https://github.com/open-mmlab/mmengine/blob/2135d3703ebffad6097c78cc4ce04aac2124360d/mmengine/optim/optimizer/builder.py#L149-L153

Reply:
From the perspective of code readability, explicitly skipping registration is preferable to implicitly skipping it by passing in `None`.

BTW, I'm not familiar with the optimizer in bitsandbytes, why don't we refer to the implementation here

https://github.com/open-mmlab/mmengine/blob/2135d3703ebffad6097c78cc4ce04aac2124360d/mmengine/optim/optimizer/builder.py#L120

 to register the optimizer?

Reply:
Me neither. I just attempted to fix the bug with minimal changes. I'm uncertain if modifying the builder function for `bitsandbytes` to resemble `Sophia`'s could cause any issues